web/packages/api/src/toPolkadotSnowbridgeV2.ts

```diff
@@ -0,0 +1,330 @@
+import { TransferInterface } from "./transfers/toPolkadot/transferInterface"
+import { ERC20ToAH } from "./transfers/toPolkadot/erc20ToAH"
+import { Asset, AssetRegistry, ERC20Metadata, Parachain } from "@snowbridge/base-types"
+import { PNAToAH } from "./transfers/toPolkadot/pnaToAH"
+import { ERC20ToParachain } from "./transfers/toPolkadot/erc20ToParachain"
+import { PNAToParachain } from "./transfers/toPolkadot/pnaToParachain"
+import { MultiAddressStruct } from "@snowbridge/contract-types/dist/IGateway.sol/IGatewayV1"
+import {
+    AbiCoder,
+    AbstractProvider,
+    ContractTransaction,
+    LogDescription,
+    TransactionReceipt,
+} from "ethers"
+import { hexToU8a, stringToU8a } from "@polkadot/util"
+import { blake2AsHex } from "@polkadot/util-crypto"
+import { IERC20__factory, IGatewayV2__factory } from "@snowbridge/contract-types"
+import { ParachainBase } from "./parachains/parachainBase"
+import { OperationStatus } from "./status"
+import { FeeInfo, ValidationLog } from "./toPolkadot_v2"
+import { ApiPromise } from "@polkadot/api"
+import { Result } from "@polkadot/types"
+import { XcmDryRunApiError, XcmDryRunEffects } from "@polkadot/types/interfaces"
+import { accountToLocation } from "./xcmBuilder"
+export { ValidationKind } from "./toPolkadot_v2"
+
+export type DeliveryFee = {
+    assetHubDeliveryFeeEther: bigint
+    assetHubExecutionFeeEther: bigint
+    destinationDeliveryFeeEther: bigint
+    destinationExecutionFeeEther: bigint
+    relayerFee: bigint
+    totalFeeInWei: bigint
+}
+
+export type Transfer = {
+    input: {
+        registry: AssetRegistry
+        sourceAccount: string
+        beneficiaryAccount: string
+        tokenAddress: string
+        destinationParaId: number
+        amount: bigint
+        fee: DeliveryFee
+    }
+    computed: {
+        gatewayAddress: string
+        beneficiaryAddressHex: string
+        beneficiaryMultiAddress: MultiAddressStruct
+        totalValue: bigint
+        tokenErcMetadata: ERC20Metadata
+        ahAssetMetadata: Asset
+        destAssetMetadata: Asset
+        destParachain: Parachain
+        minimalBalance: bigint
+        topic: string
+    }
+    tx: ContractTransaction
+}
+
+export type ValidationResult = {
+    logs: ValidationLog[]
+    success: boolean
+    data: {
+        etherBalance: bigint
+        tokenBalance: {
+            balance: bigint
+            gatewayAllowance: bigint
+        }
+        feeInfo?: FeeInfo
+        bridgeStatus: OperationStatus
+        assetHubDryRunError?: string
+        destinationParachainDryRunError?: string
+    }
+    transfer: Transfer
+}
```

Comment:
It seems to be the same data structure as in the legacy toPolkadot_v2. Could we just reuse it?

Reply:
The `Transfer` type here is the type defined in `toPolkadotSnowbridgeV2`, which is why I could not reuse the `ValidationResult` type in the legacy `toPolkadot_v2`.